### PR TITLE
p2p fixes

### PR DESCRIPTION
**Pull request by dkuehr**
- protocol select: improve/fix timeouts
- pnet: implement timeouts for nonce exchange
- identify: close streams
- pubsub: close streams

### CODE REVIEW

File: p2p/src/network/identify/p2p_network_identify_state.rs [hunk 1]
```diff
@@ -39,4 +39,8 @@ impl P2pNetworkIdentifyState {
             .get_mut(peer_id)
             .map_or(false, |m| m.remove(stream_id).is_some())
     }
+
+    pub fn remove_peer(&mut self, peer_id: &PeerId) {
```
Comment (akoptelov) [inline p2p/src/network/identify/p2p_network_identify_state.rs:43]: Let's give it a name that better describes its functionality.

File: p2p/src/p2p_config.rs [hunk 2]
```diff
@@ -110,6 +111,7 @@ impl Default for P2pTimeouts {
                 Some(Duration::from_secs(5)),
             ),
             select: from_env_or("SELECT_TIMEOUT", Some(Duration::from_secs(5))),
+            pnet: Some(Duration::from_secs(5)),
```
Comment (akoptelov) [inline p2p/src/p2p_config.rs:114]: Please make it configurable via env. BTW, isn't 5 secs too much for PNET?

**Review by akoptelov — DISMISSED** [review]
LGTM, with some minor changes.

File: p2p/src/network/identify/p2p_network_identify_state.rs (hunk 1 above)
Comment (dkuehr) [reply p2p/src/network/identify/p2p_network_identify_state.rs:43]: Would `purge_peer_state` do?

File: p2p/src/p2p_config.rs (hunk 2 above)
Comment (dkuehr) [reply p2p/src/p2p_config.rs:114]: Sure, I'll make it configurable via env var like you suggest.
Five secs is probably too much but I wanted to be on the safe side for some of the tests, maybe we can reduce the default value now and increase it via env var for the tests.

**Pull request merged**